drivers/soundwire/bus.c

```diff
@@ -1799,6 +1799,25 @@ int sdw_handle_slave_status(struct sdw_bus *bus,
 		    slave->status != SDW_SLAVE_UNATTACHED) {
 			dev_warn(&slave->dev, "Slave %d state check1: UNATTACHED, status was %d\n",
 				 i, slave->status);
+
+			if (bus->recheck_unattached && bus->ops->read_ping_status) {
+				u32 ping_status;
+
+				mutex_lock(&bus->msg_lock);
+
+				ping_status = bus->ops->read_ping_status(bus);
```

Comment:
I am thinking if we will get the same result when we read ping status and from `cdns_update_slave_status`. If so, we may still see the issue if the ping status is still UNATTACHED (and will become ATTACHED later) when we read it here.

Reply:
@bardliao if the ping status is UNATTACHED, then the device should re-appear as Device0 and the existing logic for the enumeration should work.

What doesn't work is when a device is reported as UNATTACHED, but that's a false positive. In that case, the device will NEVER report as Device0 and will never be re-enumerated again. The logic will wait for something that will not happen and times out.